controllers/backupstorage_controller.go

```diff
@@ -97,35 +103,78 @@ func (r *BackupStorageReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 		return ctrl.Result{}, err
 	}
-	var needsUpdate bool
-	if req.NamespacedName.Namespace == r.systemNamespace {
-		logger.Info("setting controller references for the secret")
-		needsUpdate, err = r.reconcileBackupStorage(ctx, bs, defaultSecret)
-		if err != nil {
+
+	if req.NamespacedName.Namespace != r.systemNamespace {
+		return ctrl.Result{}, nil
+	}
+
+	// Set controllerRef on the defaultSecret.
+	if !metav1.IsControlledBy(defaultSecret, bs) {
+		logger.Info("setting controller reference for the secret")
+		if err := controllerutil.SetControllerReference(bs, defaultSecret, r.Scheme); err != nil {
 			return ctrl.Result{}, err
 		}
-		if !controllerutil.ContainsFinalizer(bs, cleanupSecretsFinalizer) {
-			controllerutil.AddFinalizer(bs, cleanupSecretsFinalizer)
-			if err := r.Update(ctx, bs); err != nil {
-				return ctrl.Result{}, err
-			}
+		if err := r.Update(ctx, defaultSecret); err != nil {
+			return ctrl.Result{}, err
 		}
 	}
 
-	logger.Info("updating secrets in used namespaces")
-	if err := r.handleSecretUpdate(ctx, bs, defaultSecret); err != nil {
+	// Add finalizer.
```

Comment:
I don't think this comment adds anything :sweat_smile: the function is literally called `AddFinalizer` :see_no_evil: 
```suggestion
```

Reply:
Sorry, I have the habit of leaving placeholder comments everywhere and then forgetting to remove it. :)